Failed to build if PoissonReconstruction enabled in VTK build

From @lorensen  :  

I have enabled `PoissonReconstruction` in my vtk build, but the VTKExample that uses it
gets the following error( It worked before @jcfr VTK changes)

```
$ make PoissonExtractSurface
Building CXX object
src/Cxx/Points/CMakeFiles/PoissonExtractSurface.dir/PoissonExtractSurface.cxx.o
/Users/lorensen/ProjectsGIT/VTKExamples/src/Cxx/Points/PoissonExtractSurface.cxx:9:10:
fatal error:
      'vtkPoissonReconstruction.h' file not found
#include <vtkPoissonReconstruction.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[3]: *** [src/Cxx/Points/CMakeFiles/PoissonExtractSurface.dir/PoissonExtractSurface.cxx.o]
Error 1
make[2]: *** [src/Cxx/Points/CMakeFiles/PoissonExtractSurface.dir/all] Error 2
make[1]: *** [src/Cxx/Points/CMakeFiles/PoissonExtractSurface.dir/rule] Error 2
make: *** [PoissonExtractSurface] Error 2
```

I have a fix. In the past I did not need to include the module explicitly.
Now I do. That's OK.

I'll push a change.


On Sat, Oct 21, 2017 at 7:36 PM, Jean-Christophe Fillion-Robin <
notifications@github.com> wrote:

> From @lorensen <https://github.com/lorensen> :
>
> I have enabled PoissonReconstruction in my vtk build, but the VTKExample
> that uses it
> gets the following error( It worked before @jcfr <https://github.com/jcfr>
> VTK changes)
>
> $ make PoissonExtractSurface
> Building CXX object
> src/Cxx/Points/CMakeFiles/PoissonExtractSurface.dir/PoissonExtractSurface.cxx.o
> /Users/lorensen/ProjectsGIT/VTKExamples/src/Cxx/Points/PoissonExtractSurface.cxx:9:10:
> fatal error:
>       'vtkPoissonReconstruction.h' file not found
> #include <vtkPoissonReconstruction.h>
>          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 1 error generated.
> make[3]: *** [src/Cxx/Points/CMakeFiles/PoissonExtractSurface.dir/PoissonExtractSurface.cxx.o]
> Error 1
> make[2]: *** [src/Cxx/Points/CMakeFiles/PoissonExtractSurface.dir/all] Error 2
> make[1]: *** [src/Cxx/Points/CMakeFiles/PoissonExtractSurface.dir/rule] Error 2
> make: *** [PoissonExtractSurface] Error 2
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/lorensen/VTKExamples/issues/68>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAUFDNqscyF1CxxHqH15NtBZdVBJUMLpks5suqokgaJpZM4QByjo>
> .
>



-- 
Unpaid intern in BillsBasement at noware dot com

--- a/CMake/WikiModuleConfig.cmake
+++ b/CMake/WikiModuleConfig.cmake
@@ -27,6 +27,14 @@ endif()
 # Load up the CMake variables we need.
 vtk_module_export_info()
 
+# If available, load optional VTK modules.
+foreach(optional_dep IN LISTS WikiExamples_OPTIONAL_DEPENDS)
+  list(FIND VTK_MODULES_AVAILABLE ${optional_dep} found)
+  if (found GREATER -1)
+    list(APPEND ${vtk-module}_DEPENDS ${optional_dep})
+  endif()
+endforeach()
+
 # Load requested modules and their dependencies into
 # variables VTK_(DEFINITIONS|LIBRARIES|INCLUDE_DIRS|LIBRARY_DIRS).
 vtk_module_config(VTK ${${vtk-module}_DEPENDS})

--- a/module.cmake
+++ b/module.cmake
@@ -50,30 +50,27 @@ set(DEPENDS
   vtkViewsInfovis
   )
 
-if(${Powercrust_LOADED})
-  list(APPEND DEPENDS
-    Powercrust
+set(WikiExamples_OPTIONAL_DEPENDS
+  PoissonReconstruction
+  Powercrust
+  SplineDrivenImageSlicer
+  vtkFiltersParallelStatistics
+  vtkGUISupportQtOpenGL
+  vtkInfovisBoostGraphAlgorithms
+  vtkIOFFMPEG
+  vtkIOMySQL
+  vtkRenderingQt
+  vtkViewsQt
   )
-endif()
 
-if(${SplineDrivenImageSlicer_LOADED})
-  list(APPEND DEPENDS
-    SplineDrivenImageSlicer
-  )
-endif()
-
-if(${PoissonReconstruction_LOADED})
-  list(APPEND DEPENDS
-    PoissonReconstruction
-  )
-endif()
-
-if(Module_vtkGUISupportQtOpenGL)
-  list(APPEND DEPENDS
-    vtkGUISupportQtOpenGL
-    vtkRenderingQt
-    vtkViewsQt
-    )
+if(${VTK_VERSION} VERSION_LESS "8.1")
+  foreach(optional_dep IN LISTS WikiExamples_OPTIONAL_DEPENDS)
+    if(TARGET ${optional_dep} OR ${optional_dep}_LOADED OR Module_${optional_dep})
+      list(APPEND DEPENDS
+        ${optional_dep}
+        )
+    endif()
+  endforeach()
 endif()
 
 vtk_module( WikiExamples

